### DISCUSSION TITLE
I’m creating a React-native project using Supabase, i am stuck on implementing in-app and push notification

### DISCUSSION THREAD

**Discussion by emmyduruc** [supabase/supabase#19321] [Questions]
I know in Firebase it is quite easy using Notifee in supabase how do i go about it i have seen some suggestions about using onesignal but that solution does not solve my criteria because its web base.

Can anyone assist me on how to successfully have a push notification in my app

**Comment by GaryAustin1** [comment]
The onesignal solution I've seen uses Edge Functions to do the sending.  So your app would call an edge function or you could have a table in the database you insert to that then uses a webhook to call the edge function.
https://supabase.com/partners/integrations/onesignal

**Reply by emmyduruc** [reply]
> The onesignal solution I've seen uses Edge Functions to do the sending.  So your app would call an edge function or you could have a table in the database you insert to that then uses a webhook to call the edge function.
> 
> https://supabase.com/partners/integrations/onesignal

Thanks @GaryAustin1 for your response going through Onesignal and having to create a Firebase project to get the json required from Onesignal is annoying in this case i wish there's a way to handle it better that i won't have to create a Firebase project for this.

**Reply by GaryAustin1** [reply]
Sorry I've never done it.  I just skimmed and see that appears to be required to send to Android devices.

**Reply by GaryAustin1** [reply]
I found this comment in one discussion....
`I dont really think Supabase can ever have an alternative due to Android relying on FCM in the background to send all notifications on the OS. Any solution that Supabase can come up with will require FCM anyways. Same with OneSignal, who also has to use FCM to send notifications to Android users.
`

**Reply by bielcitech** [reply]
In this case, I developped the edge function and managed with webhooks, but it's sending notification to all users. What I want is to send notifications to specific users depending on their categories.

Can someone help me with this king of Edge function ?
